dask_saturn/core.py

```diff
@@ -1,92 +1,78 @@
 import os
 import requests
 import json
+import asyncio
+import weakref
```

Comment:
TIL `weakref`, cool!

Reply:
Yeah it's an odd duck. I just learned about it too and this PR doesn't really use it for much. But it could be a good way to let people know that they cluster that they are connected to is already in use.

Reply:
use these all the time in game dev for passing object references around so you can detect if something still exists or has been destroyed.